src/Silhouette.js

```diff
@@ -78,9 +103,45 @@ class Silhouette {
          */
         this._colorData = null;
 
+        /**
+         * Whether or not the color data is premultiplied with its alpha channel.
+         * If it isn't, it will be multiplied here.
+         * @type {boolean}
+         */
+        this._isPremultiplied = false;
+
+        // By default, silhouettes are assumed not to contain premultiplied image data,
+        // so when we get a color, we want to multiply it by its alpha channel.
+        // Point `_getColor` to the version of the function that multiplies.
+        this._getColor = getColor4b;
+
         this.colorAtNearest = this.colorAtLinear = (_, dst) => dst.fill(0);
     }
 
+    /**
+     * @returns {boolean} true if the silhouette color data is premultiplied, false if not.
+     */
+    get premultiplied () {
+        return this._isPremultiplied;
+    }
+
+    /**
+     * Set the alpha premultiplication state of this silhouette, to ensure proper color values are returned.
+     * If set to true, the silhouette will assume it is being set with premultiplied color data,
+     * and will not multiply color values by alpha.
+     * If set to false, it will multiply color values by alpha.
+     * @param {boolean} isPremultiplied Whether this silhouette will be populated with premultiplied color data.
+     */
+    set premultiplied (isPremultiplied) {
```

Comment:
It might make sense to add this as a parameter to the `update()` method since it's a property of the pixel data, not really a property of the silhouette as seen from the "outside" -- the silhouette always tries to return premultiplied color data.